service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/LagomConfigComponent.scala

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.api
+
+import com.typesafe.config.Config
+import play.api.Configuration
+
+trait LagomConfigComponent {
+  def configuration: Configuration
+  def config: Config = configuration.underlying
+}
```

Comment:
Do you mean using `play.api.Configuration` everywhere instead of `play.Configuration` or `com.typesafe.config.Config`?

The java `play.Configuration` is now deprecated with the following message:
> As of release 2.6.0. Use {@link Config} instead.
I think this was the trigger for us to rely on `Config` as much as we can. 

However, the scala `play.api.Configuration` is still around in play with some deprecated methods. For the deprecated methods, the recommendation is to fallback to the underlying `Config` object.

There is nothing in `play.api.Configuration` that is being used by Lagom. The only reason we are using it is because we get it from Play. Hence the idea of a new trait component to take the Play one and expose the underlying `Config` to Lagom.